src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -173,7 +174,18 @@ static ucs_status_t uct_cuda_ipc_rkey_unpack(uct_component_t *component,
 static ucs_status_t uct_cuda_ipc_rkey_release(uct_component_t *component,
                                               uct_rkey_t rkey, void *handle)
 {
+    uct_cuda_ipc_key_t *key = (uct_cuda_ipc_key_t *) rkey;
+
     ucs_assert(NULL == handle);
+
+    if ((key->unmap_memhandle_func != NULL) && (key->d_mapped != 0)) {
```

Comment:
@pentschev @bureddy by releasing memhandle during release key, the problem of memory overhead is further exacerbated, right? If I'm not wrong, release key occurs during finalize. Correct me if I'm mistaken

Reply:
> by releasing memhandle during release key, the problem of memory overhead is further exacerbated, right?

I'm not sure I understood your question correctly, but I think what you're saying is that if we could release it earlier, then we would free memory that isn't going to be used anymore, is that right?

> If I'm not wrong, release key occurs during finalize. Correct me if I'm mistaken

I'm not sure exactly what trigger the release of the key. However, the key maps one memory block and we are only allowed to have one mapping per memory block, so we can't close it before the rkey can be released.